smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/providers/connectors/WorkerPoolRegistry.java

```diff
@@ -54,7 +55,11 @@ public void terminate(
             @Observes(notifyObserver = Reception.IF_EXISTS) @Priority(100) @BeforeDestroyed(ApplicationScoped.class) Object event) {
         if (!workerExecutors.isEmpty()) {
             for (WorkerExecutor executor : workerExecutors.values()) {
-                executor.close();
+                if (Infrastructure.canCallerThreadBeBlocked()) {
```

Comment:
I didn't know that method

Reply:
It's from Mutiny. When you integrate Mutiny into your framework/server, you need to provide the proper implementation.

Reply:
By default, it's already true - but it will be different in Quarkus.